backend/sites/src/app/services/folio/folio.service.ts

```diff
@@ -0,0 +1,202 @@
+        where: { userId, folioId },
+      });
+
+      if (existingRecord) {
+        return false;
+      } else {
+        const folio = plainToInstance(Folio, inputDTO);
+        folio.whenCreated = new Date();
+        folio.whenUpdated = new Date();
+        const result = await this.folioRepository.save(folio);
+
+        if (result) return true;
+
+        return false;
+      }
+    } catch (error) {
+      console.error('Error', error);
+      throw new Error('Failed to add folio content');
+    }
+  }
+
+  async updateFolio(inputArr: FolioDTO[], userInfo: any): Promise<boolean> {
+    try {
+      const userId = userInfo.sub;
+      const updatePromises = inputArr.map(async (inputDTO) => {
+        const { id, description, folioId } = inputDTO;
+
+        const existingRecord = await this.folioRepository.findOne({
+          where: { id, userId },
+        });
+
+        if (existingRecord) {
+          existingRecord.description = description;
+          existingRecord.folioId = folioId;
+          existingRecord.whenUpdated = new Date();
+
+          const result = await this.folioRepository.save(existingRecord);
+
+          if (!result) {
+            console.error(`Failed to save record with id ${id}`);
+          }
+        }
+      });
+
+      await Promise.all(updatePromises);
+    } catch (error) {
+      console.error('Error', error);
+      throw new Error('Failed to update folio content');
+    }
+
+    return true;
+  }
+
+  async deleteFolio(id: number, userInfo: any): Promise<boolean> {
+    try {
+      if (id <= 0) {
+        return false;
+      }
+
+      const userId = userInfo.sub;
+
+      const savedFolio = this.folioRepository.findOne({
+        where: { id, userId },
+      });
+
+      if (savedFolio != null) {
+        await this.folioContentService.deleteAllSitesInFolio(id.toString());
+        const result = await this.folioRepository.delete({ id });
+        return result.affected > 0;
+      }
+
+      return false;
+    } catch (error) {
+      console.error('Error in deleteFolio:', error);
+      return false;
+    }
+  }
+
+  async addSiteToFolio(
+    inputDTO: [FolioContentDTO],
+    userInfo: any,
+  ): Promise<boolean> {
+    try {
+      if (inputDTO.length > 0) {
+        inputDTO.forEach((item) => {
+          if (item != null) {
+            const userId = userInfo.sub;
+            const { id } = item;
+            this.folioRepository
+              .findOne({ where: { id, userId } })
+              .then((folio) => {
+                if (folio) {
+                  item.folioId = folio.id.toString();
+                  this.folioContentService.addFolioContent(item);
+                }
+              });
+          }
+        });
+      }
+
+      return true;
+    } catch (error) {
+      console.error('Error', error);
+      throw error;
+    }
+  }
+
+  async deleteSitesInFolio(
+    inputDTO: [FolioContentDTO],
+    userInfo: any,
+  ): Promise<boolean> {
```

Comment:
can we use bulk delete ?

Reply:
No delete from this block , separate service for folio content invoked from here.